addons/website/static/tests/tours/edit_megamenu.js

```diff
@@ -92,4 +92,83 @@ tour.register('edit_megamenu', {
         run: function () {}, // it's a check
     },
 ]);
+tour.register('edit_megamenu_big_icons_subtitles', {
+    test: true,
+    url: '/?enable_editor=1',
+}, [
+    // Add a megamenu item to the top menu.
+    {
+        content: "Click on a menu item",
+        trigger: '#top_menu .nav-item a',
+    },
+    {
+        content: "Click on 'Link' to open Link Dialog",
+        trigger: '.o_edit_menu_popover a.js_edit_menu',
+    },
+    {
+        content: "Trigger the link dialog (click 'Add Mega Menu Item')",
+        extra_trigger: ".o_web_editor_dialog",
+        trigger: '.modal-body a.js_add_menu[data-type="mega"]',
+    },
+    {
+        content: "Write a label for the new menu item",
+        trigger: '.o_link_dialog #o_link_dialog_label_input',
+        run: 'text Megaaaaa2!',
+    },
+    {
+        content: "Save the new menu item",
+        trigger: '.modal-footer .btn-primary',
+    },
+    {
+        content: "Save the changes to the menu",
+        trigger: '.modal-footer .btn-primary',
+    },
+    wTourUtils.clickOnExtraMenuItem(),
+    {
+        content: "Check for the new mega menu",
+        trigger: '#top_menu .nav-item a.o_mega_menu_toggle:contains("Megaaaaa2!")',
+        run: function () {}, // it's a check
+    },
```

Comment:
Followup of https://github.com/odoo/odoo/pull/145925/files#r1464527216
I had a deeper look, right now the modal is actually not saved when you click on it: it calls for a page reload while remaining open, meaning elements behind it won't be considered with normal trigger (they are considered hidden).
So there was probably no race condition possible anyway.. but only while we keep this "keep open" behavior. So it's still best to consider that the popup could be closed on "Save" click and prevent race condition, instead of not considering it and making it break the day we change that (easy win before it even happens).

> rdeodoo Is it ok now?

So the answer is not really: those steps will still be "true" directly after clicking save (before the page is actually reloaded, and considering the case of the modal closing on click).
It's generally how you can find race condition:
```js
{
    content: "Something",
    trigger: '.element1',
}, {
    content: "Something else",
    trigger: '.element2,
},
```
If you do that, you are not waiting that the behavior which happened after clicking on element1 is finished before clicking on element2 -> Race condition everyday.

It's what is being done here (well, not really, since the modal doesn't close on save, but you got it):
```js
{
    content: "Save the changes to the menu",
    trigger: '.modal-footer .btn-primary',
},
wTourUtils.clickOnExtraMenuItem(),
{
    content: "Check for the new mega menu",
    trigger: '#top_menu .nav-item a.o_mega_menu_toggle:contains("Megaaaaa2!")',
    run: function () {}, // it's a check
},
```
- You click on save, which fires a rpc and then a page reload when the rpc is back
- Right after having click on save, you click on extra menu item, without waiting the rpc/page reload, that trigger is immediately true/found

With the current state of things, it is exactly doing the same as before: still the same room for race condition should we make the click on save close the modal. But since it's not closing it, it's ignoring the mega menu item "hidden" behind the modal, and it's actually waiting the page reload "by chance" (because we don't close the modal).

If we were to hide the modal on click (and then wait for the rpc to reload), it would actually click on extra menu item, then wait for the "Megaaaa2" items which doesn't exist since the page is not yet reloaded. Then the page would reload, and the tour would be stuck (because Megaaa2 would be in the DOM now but the mega menu would not be opened, as the mega menu opening step was wrongly done before the page reload).

At the end, I guess it should should be about

```suggestion
    {
        content: "Save the changes to the menu",
        trigger: '.modal-footer .btn-primary',
    },
    {
        content: "Check for the new mega menu",
        trigger: '#top_menu:has(.nav-item a.o_mega_menu_toggle:contains("Megaaaaa2!"))',
        run: function () {}, // it's a check
    },
    wTourUtils.clickOnExtraMenuItem(),
```
This `has()` should really wait correctly the page reload, and will workaround the fact it's a hidden element which we want to be used in a trigger step (as #top_menu is visible, it will work).

To be confirmed, just wrote that on github.

Note: since I figured after having a deeper look that there is no race condition possible thanks to the modal not closing itself on "save" click, you can ignore this if you prefer. I still wanted to take time to explain you how those race conditions works.

Reply:
Thanks for the explanation!!! :)